docs/user_docs/concepts/package-and-module.md

```diff
@@ -0,0 +1,154 @@
+
+> Note: The use of a KCL module e.g., `kcl.mod` is optional, but required if one wants to manage, distribute, share and reuse code with a semantic version.
+
+### Creating a module
+
+A module can be created by running the following command within the module root:
+
+```bash
+kpm init [module name]
+```
+
+The module name is **required** if a package within the module needs to import another package within the module. A module can also be created by setting up the `kcl.mod` file manually.
+
+## Package
+
+In KCL, a package is usually composed of a "folder" containing KCL files. This folder can be a real disk physical path, or it can be composed of multiple KCL files (usually main package). Different packages are uniquely located by different package paths (such as `kubernetes.core.v1`)
+
+Within the same module, different packages can be imported from each other through the import statement of relative or absolute path. In the KCL, the relative import will be replaced by absolute import and the corresponding KCL code will be found through the package path.
+
+### Relative Import Path
+
+We can use the operator `.` to realize the relative path import of KCL entry files.
+
+main.k:
+
+```python
+import .model1  # Current directory module
+import ..service  # Parent directory
+import ...root  # Parent of parent directory
+
+s = service.ImageService {}
+m = root.Schema {}
+```
+
+### Absolute Import Path
+
+The semantics of `import a.b.c.d` is
+
+1. If `kcl.mod` not exist, search the path `./a/b/c/d` from the current directory.
+2. If the current directory search fails, search from the root path `ROOT_PATH/a/b/c/d`, else raise a import error.
+
+The definition of the root path `ROOT_PATH` is the directory corresponding to the `kcl.mod` file from the current directory.
+
+Code structure:
+
+```
+. 
+└── root
+    ├── kcl.mod
+    ├── model
+    │   ├── model1.k
+    |   ├── model2.k
+    │   └── main.k
+    ├── service
+    │   └── service1.k
+    └── mixin
+        └── mixin1.k
+```
+
+### Builtin Package
+
+KCL has a collection of builtin packages such as `math`, `regex`, etc. To use a builtin package, import it directly and invoke the functions using its qualified identifier. For instance,
+
+```python
+import regex
+
+image = "nginx:1.14.2"
+is_match = regex.match(image, "^[a-zA-Z]+:\d+\.\d+\.\d+$")
+
+```
+
+The output YAML is
+
+```yaml
+image: nginx:1.14.2
+is_match: true
+```
+
+### Plugin Package
+
+KCL also has a collection of plugin packages such as `hello`, `project_context`, etc. To use a plugin package, import it with a `kcl_plugin.` package path prefix and invoke the functions using its qualified identifier. For instance,
```

Comment:
the `hello` package seems not so typical to be listed here as an example of plugin packages.